src/hash_map.h

```diff
@@ -50,7 +50,22 @@ unsigned int MurmurHash2(const void* key, size_t len) {
   return h;
 }
 
-#ifdef _MSC_VER
+#if (__cplusplus >= 201103L) || (_MSC_VER >= 1900)
+#include <unordered_map>
+
+namespace std {
+template<>
+struct hash<StringPiece> {
+  typedef StringPiece argument_type;
+  typedef std::size_t result_type;
```

Comment:
Another question: I was looking at the C++ standard, hoping that result_type and argument_type aren't necessary (libc++ doesn't need them, but it looks like the standard asks for them :-/), and it says "the instantiation hash<Key> shall … satisfy the Hash requirements (17.6.3.4), with Key as the function call argument type". Key is StringPiece here, so should the parameter be per value instead of per const reference? StringPiece is small and can be passed by value efficiently, so that's not a concern.